src/Public/LogRhythm/Admin/Get-LRListGuidByName.ps1

```
@@ -27,34 +27,23 @@ Function Get-LRListGuidByName {
 
     [CmdletBinding()]
     Param(
-        [Parameter(Mandatory=$true,Position=0)]
+        [Parameter(Mandatory = $false, Position = 0)]
         [ValidateNotNull()]
-        [pscredential] $Credential,
+        [pscredential] $Credential = $SrfPreferences.LrDeployment.LrApiToken,
 
         [Parameter(Mandatory=$true,Position=1, ValueFromPipeline=$true)]
         [ValidateNotNullOrEmpty()]
         [string] $Name
     )
 
     Begin {
-        $BaseUrl = $SrfPreferences.LRDeployment.AdminApiBaseUrl
-        $Method = $HttpMethod.Get
-        $Token = $Credential.GetNetworkCredential().Password
     }
 
 
     Process {
-        Write-Verbose $BaseUrl
-
-        ## Script API Setup
-        $Token = $Credential.GetNetworkCredential().Password
-        $Headers = [Dictionary[string,string]]::new()
-        $Headers.Add("Authorization", "Bearer $Token")
-        $Headers.Add("name", $Name)
-        $RequestUrl = $BaseUrl + "/lists/"
 
         try {
-            $Response = Invoke-RestMethod -Uri $RequestUrl -Headers $Headers -Method $Method
+            $Response = Get-LrLists -Name $Name
         }
         catch [System.Net.WebException] {
             $Err = Get-RestErrorMessage $_
```

Because we aren't directly making a Web/Rest request in the function anymore (line 46), we shouldn't use `Get-RestErrorMessage` in the catch block - we should just re-throw the exception from `Get-LrLists`.

Please update the catch block to:

```
$PSCmdlet.ThrowTerminatingError($PSItem)
```

---

Yep, I figured that would be the case.  Simple enough change!